Lib/unittest/test/testmock/testmock.py

```diff
@@ -1413,6 +1413,24 @@ def test_create_autospec_with_name(self):
         m = mock.create_autospec(object(), name='sweet_func')
         self.assertIn('sweet_func', repr(m))
 
+    #Issue23078
+    def test_create_autospec_classmethod(self):
+        class TestClass:
+            method = classmethod(lambda cls: None)
+        mock_bound_meth = mock.create_autospec(TestClass.method)
+        mock_bound_meth()
+        mock_bound_meth.assert_called_once_with()
+        self.assertRaises(TypeError, mock_bound_meth, 'extra_arg')
+
+    #Issue23078
+    def test_create_autospec_staticmethod(self):
+        class TestClass:
+            method = staticmethod(lambda: None)
```

Comment:
It might be easier to read with the common syntax decorating the method.

Reply:
Thanks, changed to decorators 👍